test_runner/regress/test_tenants.py

```diff
@@ -351,6 +362,11 @@ def get_ps_metric_samples_for_timeline(
             state=TimelineArchivalState.ARCHIVED,
         )
         env.pageserver.http_client().timeline_offload(tenant_1, timeline)
+        # We need to wait until all background jobs are finished before we can check the metrics.
+        # There're many of them: compaction, GC, etc. It's hard to wait for all of them, so we just
+        # sleep for a while there.
+        time.sleep(5)
```

Comment:
I think it would be better to do wait_until instead of sleeping with a constant, sleeping with a constant is an anti pattern (tests might execute more slowly)